eventhubs/data-plane/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/OperationResultBase.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.microsoft.azure.eventhubs.impl;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+
+class OperationResultBase<T, E extends Exception> implements OperationResult<T, E> {
+    private final Consumer<T> onComplete;
+    private final Consumer<Exception> onError;
+
+    OperationResultBase(Consumer<T> onComplete, Consumer<Exception> onError) {
+        Objects.requireNonNull(onComplete);
+        Objects.requireNonNull(onError);
+
+        this.onComplete = onComplete;
+        this.onError = onError;
+    }
+
+    @Override
+    public void onComplete(T result) {
+        onComplete.accept(result);
+    }
+
+    @Override
+    public void onError(E error) {
+        onError.accept(error);
+    }
+}
```

Comment:
nit: new line is missing. I think I forgot to add it.

Reply:
Odd. it should have a new line at the end of file since my local has it.